server/server.go

```diff
@@ -706,7 +706,7 @@ func (a *ArgoCDServer) watchSettings() {
 			log.Infof("gogs secret modified. restarting")
 			break
 		}
-		if prevExtConfig != a.settings.ExtensionConfig {
+		if !equalExtConfig(prevExtConfig, a.settings.ExtensionConfig) {
```

Comment:
Maybe just reflect.DeepEqual? Less code for us, their implementation looks pretty simple. Don't think it's performance-critical code.

Reply:
fixed